src/Illuminate/Queue/RedisQueue.php

```diff
@@ -367,7 +375,24 @@ public function getQueue($queue)
      */
     public function getConnection()
     {
-        return $this->redis->connection($this->connection);
+        if (isset($this->redisConnection)) {
+            return $this->redisConnection;
+        }
+
+        $this->redisConnection = $this->redis->connection($this->connection);
+
+        if ($this->redisConnection instanceof PhpRedisConnection) {
+            $this->redisConnection = $this->redis->resolve($this->connection);
+
+            if (defined('\Redis::OPT_COMPRESSION') && $this->redisConnection->client()->getOption(\Redis::OPT_COMPRESSION) !== \Redis::COMPRESSION_NONE) {
+                $this->redisConnection->client()->setOption(\Redis::OPT_COMPRESSION, \Redis::COMPRESSION_NONE);
+            }
+            if (defined('\Redis::OPT_SERIALIZER') && $this->redisConnection->client()->getOption(\Redis::OPT_SERIALIZER) !== \Redis::SERIALIZER_NONE) {
```

Comment:
`OPT_SERIALIZER` is always defined, so is `SERIALIZER_NONE` and `COMPRESSION_NONE`.